### PR TITLE
user-workbench

### PR DESCRIPTION
**Pull request by chengjoey**
#### Which issue(s) this PR fixes:

Fixes #

#### Specified Reivewers:
/assgin @your-reviewer

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation usage docs, etc.:

```docs

```

### CODE REVIEW

File: modules/cmdb/dao/application.go [hunk 1]
```diff
@@ -171,3 +172,12 @@ func (client *DBClient) GetAllApps() ([]model.Application, error) {
 	}
 	return applications, nil
 }
+
+// GetJoinedAppNumByUserId 获取用户在某个企业下的参与应用数
```
Comment (sfwn) [inline modules/cmdb/dao/application.go:176]: All new comments use English please.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: bundle/workbench.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package bundle
+
+import (
+	"fmt"
+	"github.com/erda-project/erda/apistructs"
```
Comment (sfwn) [inline bundle/workbench.go:5]: adjust the import order.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/cmdb/dao/application.go [hunk 3]
```diff
@@ -14,6 +14,7 @@
 package dao
 
 import (
+	"fmt"
```
Comment (kakj-go) [inline modules/cmdb/dao/application.go:17]: import

File: modules/cmdb/dao/issue_state.go [hunk 4]
```diff
@@ -123,6 +123,16 @@ func (client *DBClient) GetIssuesStatesByProjectID(projectID uint64, issueType a
 	return states, nil
 }
 
+// 增加一个用projectId列表来获取所有的状态
+func (client *DBClient) GetIssuesStatesByProjectIDList(projectIDS []uint64) ([]IssueState, error) {
```
Comment (kakj-go) [inline modules/cmdb/dao/issue_state.go:127]: projectIDS - > projectIDList

File: modules/cmdb/dao/project.go [hunk 5]
```diff
@@ -160,3 +160,46 @@ func (client *DBClient) UpdateProjectQuota(clusterName string, cpuOverSellChange
...
+	ProjectID string `json:"project_id"`
+}
+
+// GetJoinedProjectNumByUserID 根据用户和企业id查询参加的所有项目数和项目id列表
+func (client *DBClient) GetJoinedProjectNumByUserID(userID string, orgID string) (int, []string, error) {
+	var total int
+	var proIDS []ProjectID
+	res := make([]string, 0)
+	if err := client.Model(&model.Member{}).
+		Where(fmt.Sprintf("user_id = %s and org_id = %s and scope_type = \"%s\"", userID, orgID, apistructs.ProjectScopeType)).
```
Comment (kakj-go) [inline modules/cmdb/dao/project.go:174]: sql injection risk, use ? not %s

File: modules/cmdb/endpoints/endpoints.go [hunk 6]
```diff
@@ -475,7 +475,7 @@ func (e *Endpoints) Routes() []httpserver.Endpoint {
 		{Path: "/api/orgs/actions/switch", Method: http.MethodPost, Handler: e.ChangeCurrentOrg},
 		{Path: "/api/orgs/actions/relate-cluster", Method: http.MethodPost, Handler: e.CreateOrgClusterRelation},
 		{Path: "/api/orgs/clusters/relations", Method: http.MethodGet, Handler: e.ListAllOrgClusterRelation},
-		{Path: "/api/orgs/{orgID}/nexus", Method: http.MethodGet, Handler: e.GetOrgNexus},
+		{Path: "/api/orgs/{orgID}/nexus", Method: http.MethodGet , Handler: e.GetOrgNexus},
```
Comment (kakj-go) [inline modules/cmdb/endpoints/endpoints.go:478]: remove the spaces here

File: modules/cmdb/endpoints/endpoints.go [hunk 7]
```diff
@@ -515,7 +515,7 @@ func (e *Endpoints) Routes() []httpserver.Endpoint {
 		{Path: "/api/projects/actions/refer-cluster", Method: http.MethodGet, Handler: e.ReferCluster},
 		{Path: "/api/projects/actions/fill-branch-rule", Method: http.MethodGet, Handler: e.FillBranchRule},
 		{Path: "/api/projects/actions/get-project-functions", Method: http.MethodGet, Handler: e.GetFunctions},
-		{Path: "/api/projects/actions/set-project-functions", Method: http.MethodPost, Handler: e.SetFunctions},
+		{Path: "/api/projects/actions/set-xproject-functions", Method: http.MethodPost, Handler: e.SetFunctions},
```
Comment (kakj-go) [inline modules/cmdb/endpoints/endpoints.go:518]: why change project to xproject

File: modules/cmdb/endpoints/endpoints.go [hunk 8]
```diff
@@ -776,5 +776,9 @@ func (e *Endpoints) Routes() []httpserver.Endpoint {
 		{Path: "/api/reviews/actions/updateReview", Method: http.MethodPut, Handler: e.UpdateApproval},
 		{Path: "/api/reviews/actions/user/create", Method: http.MethodPost, Handler: e.CreateReviewUser},
 		{Path: "/api/reviews/actions/{taskId}", Method: http.MethodGet, Handler: e.GetReviewByTaskId},
+
+		//工作台相关
+		{Path: "/api/workbench/actions/list", Method: http.MethodGet, Handler: e.GetWorkbenchData},
+		{Path: "/api/workbench/issues/list", Method: http.MethodGet, Handler: e.GetIssuesForWorkbench},
```
Comment (kakj-go) [inline modules/cmdb/endpoints/endpoints.go:782]: /api/workbench/issues/list -> /api/workbench/issues/actions/list

File: modules/cmdb/endpoints/endpoints.go [hunk 9]
```diff
@@ -776,5 +776,9 @@ func (e *Endpoints) Routes() []httpserver.Endpoint {
 		{Path: "/api/reviews/actions/updateReview", Method: http.MethodPut, Handler: e.UpdateApproval},
 		{Path: "/api/reviews/actions/user/create", Method: http.MethodPost, Handler: e.CreateReviewUser},
 		{Path: "/api/reviews/actions/{taskId}", Method: http.MethodGet, Handler: e.GetReviewByTaskId},
+
+		//工作台相关
```
Comment (sfwn) [inline modules/cmdb/endpoints/endpoints.go:780]: english

File: modules/cmdb/endpoints/workbench.go [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+		pageReq.OrderBy = "plan_started_at"
+	case "planFinishedAt":
+		pageReq.OrderBy = "plan_finished_at"
+	case "assignee":
+		pageReq.OrderBy = "assignee"
+	default:
+		return apierrors.ErrPagingIssues.InvalidParameter("orderBy").ToResp(), nil
+	}
+
+	// 鉴权
```
Comment (sfwn) [inline modules/cmdb/endpoints/workbench.go:154]: same

File: modules/cmdb/services/issue/issue.go [hunk 11]
```diff
@@ -373,6 +373,73 @@ func (svc *Issue) Paging(req apistructs.IssuePagingRequest) ([]apistructs.Issue,
...
+		}
+		for _, v := range irs {
+			issueRelationIDs = append(issueRelationIDs, int64(v.RelatedIssue))
+		}
+	}
+	if isLabel || isIssue {
+		req.IDs = strutil.DedupInt64Slice(append(getRelatedIDs(labelRelationIDs, issueRelationIDs, isLabel, isIssue), req.IDs...))
+	}
+
+	// 该项目下全部的state信息，之后不再查询state  key: stateID value:state
```
Comment (sfwn) [inline modules/cmdb/services/issue/issue.go:420]: same

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**